xbmc/pvr/addons/PVRClient.cpp

```diff
@@ -923,6 +923,17 @@ PVR_ERROR CPVRClient::GetRecordingEdl(const CPVRRecording& recording, std::vecto
   }, m_clientCapabilities.SupportsRecordingsEdl());
 }
 
+PVR_ERROR CPVRClient::GetRecordingSize(const CPVRRecording& recording, int64_t& sizeInBytes)
+{
+  sizeInBytes = -1;
+  return DoAddonCall(__FUNCTION__, [&recording, &sizeInBytes](const AddonInstance* addon) {
+    PVR_RECORDING tag;
+    WriteClientRecordingInfo(recording, tag);
+    sizeInBytes = addon->GetRecordingSize(tag);
+    return (sizeInBytes == -1) ? PVR_ERROR_NOT_IMPLEMENTED : PVR_ERROR_NO_ERROR;
```

Comment:
This special code would not be needed if the new function would return PVR_ERROR.